src/hub.js

```diff
@@ -255,19 +148,135 @@ function mountUI(scene, props = {}) {
-    mountUI(scene, modifiedProps);
+    uiProps = { ...uiProps, ...props };
+    mountUI(scene, uiProps);
+  };
+
+  const applyProfileFromStore = playerRig => {
+    const displayName = store.state.profile.display_name;
+    playerRig.setAttribute("player-info", {
+      displayName,
+      avatarSrc: "#" + (store.state.profile.avatar_id || "botdefault")
+    });
+    document.querySelector("a-scene").emit("username-changed", { username: displayName });
+  };
+
+  const exitScene = () => {
+    hubChannel.disconnect();
+    const scene = document.querySelector("a-scene");
+    scene.renderer.animate(null); // Stop animation loop, TODO A-Frame should do this
+    document.body.removeChild(scene);
+  };
+
+  const enterScene = async (mediaStream, enterInVR, janusRoomId) => {
+    const scene = document.querySelector("a-scene");
+    const playerRig = document.querySelector("#player-rig");
+    document.querySelector("a-scene canvas").classList.remove("blurred");
+    scene.render();
+
+    if (enterInVR) {
+      scene.enterVR();
+    }
+
+    AFRAME.registerInputActions(inGameActions, "default");
+
+    document.querySelector("#player-camera").setAttribute("look-controls", "");
+
+    scene.setAttribute("networked-scene", {
+      room: janusRoomId,
+      serverURL: process.env.JANUS_SERVER
+    });
+
+    if (!qsTruthy("no_stats")) {
+      scene.setAttribute("stats", true);
+    }
+
+    if (isMobile || qsTruthy("mobile")) {
+      playerRig.setAttribute("virtual-gamepad-controls", {});
+    }
+
+    const applyProfileOnPlayerRig = applyProfileFromStore.bind(null, playerRig);
+    applyProfileOnPlayerRig();
+    store.addEventListener("statechanged", applyProfileOnPlayerRig);
+
+    const avatarScale = parseInt(qs.avatar_scale, 10);
+
+    if (avatarScale) {
+      playerRig.setAttribute("scale", { x: avatarScale, y: avatarScale, z: avatarScale });
+    }
+
+    const videoTracks = mediaStream.getVideoTracks();
+    let sharingScreen = videoTracks.length > 0;
+
+    const screenEntityId = `${NAF.clientId}-screen`;
+    let screenEntity = document.getElementById(screenEntityId);
+
+    scene.addEventListener("action_share_screen", () => {
+      sharingScreen = !sharingScreen;
+      if (sharingScreen) {
+        for (const track of videoTracks) {
+          mediaStream.addTrack(track);
+        }
+      } else {
+        for (const track of mediaStream.getVideoTracks()) {
+          mediaStream.removeTrack(track);
+        }
+      }
+      NAF.connection.adapter.setLocalMediaStream(mediaStream);
+      screenEntity.setAttribute("visible", sharingScreen);
+    });
+
+    if (!qsTruthy("offline")) {
+      document.body.addEventListener("connected", () => {
+        hubChannel.sendEntryEvent().then(() => {
+          store.update({ lastEnteredAt: moment().toJSON() });
+        });
+      });
+
+      scene.components["networked-scene"].connect().catch(connectError => {
+        // hacky until we get return codes
```

Comment:
actual change is here, other code is largely just moved around

Reply:
Looks like the actual text is "Room is full." https://github.com/mozilla/janus-plugin-sfu/blob/master/src/lib.rs#L440

Reply:
brb, putting in an error that says "network connection too full, can't send packets"

Reply:
Worth noting this is dependent on https://github.com/mozilla/naf-janus-adapter/pull/31

Reply:
Also dependent on https://github.com/mozilla/mr-social-client/pull/207